README.md
# KOGNOG LINUX FROM SCRATCH v3.0
Kognog LFS Project, based on LFS-systemd v12.0

NOTES:
======

1) You need to run this one on an Arch Linux Distro, or modify the code to run on other distro
2) For this install process to work, you need dialog installed
3) run ./install-kognoglfs.sh
4) Hope you enjoy it!

VERSION CONTROL:
================

1) 12/04/2023: First upload of KOGNOG LFS to my Github account. not sure of what I am doing so far, but here it goes. This version only runs Main Menu, READ ME, Preface, and Introduction. All sections a text extracted from the LFS Book so far, plus some of mine.


ChapterI/Changelog.txt

This is version 12.0-systemd of the Linux From Scratch book, dated September 1st, 2023. If this book is more than six months old, a newer and better version is probably already available. To find out, please check one of the mirrors via:

https://www.linuxfromscratch.org/mirrors.html


Below is a list of changes made since the previous release of the book.

Changelog Entries:
==================
Access the link below and look for the search bar (upper right side), insert the codes (ex. #5320) to read more details

https://wiki.linuxfromscratch.org/lfs/


2023-09-01

[bdubbs] - LFS-12.0 released.

2023-08-18

[bdubbs] - Update to linux-6.4.12. Fixes #5320.

2023-08-18

[bdubbs] - Update to udev-lfs-20230818.

2023-08-15

[bdubbs] - Add a patch to fix a performance regression in glibc's posix_memalign() function. Fixes #5315.

[bdubbs] - Update to less-643. Fixes #5317.

[bdubbs] - Update to meson-1.2.1. Fixes #5314.

[bdubbs] - Update to linux-6.4.10. Fixes #5313.

[bdubbs] - Update to iana-etc-20230810. Addresses #5006.

[rahul] - Update to pkgconf-2.0.1. Fixes #5316.

2023-08-07

[bdubbs] - Update to xz-5.4.4. Fixes #5307.

[bdubbs] - Update to wheel-0.41.1 (Python Module). Fixes #5311.

[bdubbs] - Update to man-pages-6.05.01. Fixes #5306.

[bdubbs] - Update to linux-6.4.8. Fixes #5309.

[bdubbs] - Update to iana-etc-20230804. Addresses #5006.

[rahul] - Update to pkgconf-2.0.0. Fixes #5310.

2023-08-01

[bdubbs] - Update to vim-9.0.1677. Addresses #4500.

[bdubbs] - Update to openssl-3.1.2. Fixes #5305.

[bdubbs] - Update to man-pages-6.05. Fixes #5303.

[bdubbs] - Update to binutils-2.41. Fixes #5300.

[bdubbs] - Update to gmp-6.3.0. Fixes #5301.

[bdubbs] - Update to glibc-2.38. Fixes #5302.

2023-07-28

[bdubbs] - Update udev-lfs tarball to remove obsolete cdrom rules and references to ISDN devices. Fixes #5291.

[bdubbs] - Update to wheel-0.41.0 (Python Module). Fixes #5290.

[bdubbs] - Update to tar-1.35. Fixes #5287.

[bdubbs] - Update to systemd-254. Fixes #5293.

[bdubbs] - Update to meson-1.2.0. Fixes #5286.

[bdubbs] - Update to linux-6.4.7. Fixes #5288.

[bdubbs] - Update to gcc-13.2.0. Fixes #5292.

[bdubbs] - Update to file-5.45. Fixes #5294.

2023-07-15

[bdubbs] - Update to iana-etc-20230629. Addresses #5006.

[bdubbs] - Update to linux-6.4.3. Fixes #5284.

[bdubbs] - Update to libxcrypt-4.4.36. Fixes #5283.

[bdubbs] - Update to groff-1.23.0. Fixes #5282.

[bdubbs] - Update to perl-5.38.0. Fixes #5281.

2023-07-02

[xry111] - Add libxcrypt-4.4.35. Fixes #5280.

[xry111] - Update to iproute2-6.4.0. Fixes #5277.

[xry111] - Update to linux-6.4.1. Fixes #5276.

2023-07-01

[bdubbs] - Update to iana-etc-20230615. Addresses #5006.

[bdubbs] - Update to vim-9.0.1671. Addresses #4500.

[bdubbs] - Update to util-linux-2.39.1. Addresses #5278.

[bdubbs] - Update to linux-6.3.10. Addresses #5276.

[rahul] - Update to kbd-2.6.1. Fixes #5279.

[bdubbs] - Update to gettext-0.22. Fixes #5275.

2023-06-17

[xry111] - Update to linux-6.3.8. Fixes #5272.

[xry111] - Update to kbd-2.6.0. Fixes #5273.

[rahul] - Changed from pkg-config to pkgconf-1.9.5. Fixes #5274.

2023-06-09

[bdubbs] - Update to dbus-1.14.8. Fixes #5271.

[bdubbs] - Update to linux-6.3.6. Fixes #5269.

[bdubbs] - Update to Python-3.11.4. Fixes #5271.

2023-06-03

[bdubbs] - Update to iana-etc-20230524. Addresses #5006.

[bdubbs] - Update to MarkupSafe-2.1.3 (Python Module). Fixes #5268.

[bdubbs] - Update to linux-6.3.5. Fixes #5264.

[bdubbs] - Update to openssl-3.1.1. Fixes #5267.

[bdubbs] - Update to meson-1.1.1. Fixes #5266.

[bdubbs] - Update to diffutils-3.10. Fixes #5262.

[bdubbs] - Update to bc-6.6.0. Fixes #5263.

2023-05-18

[bdubbs] - Update to util-linux-2.39. Fixes #5259.

[bdubbs] - Update to linux-6.3.3. Fixes #5261.

[bdubbs] - Update to libcap-2.69. Fixes #5258.

[bdubbs] - Update to grep-3.11. Fixes #5256.

[bdubbs] - Update to flit_core-3.9.0. Fixes #5257.

2023-05-13

[xry111] - Update to less-633. Fixes #5251.

[xry111] - Update to linux-6.3.2. Fixes #5255.

[xry111] - Update to xz-5.4.3. Fixes #5252.

[xry111] - Update to gawk-5.2.2. Fixes #5253.

[xry111] - Fix systemd runtime issue exploited by GCC 13. Fixes #5254.

2023-05-01

[bdubbs] - Update to vim-9.0.1503. Addresses #4500.

[bdubbs] - Update to iana-etc-20230418. Addresses #5006.

[bdubbs] - Update to iproute2-6.3.0. Fixes #5248.

[bdubbs] - Update to gcc-13.1.0. Fixes #5247.

[bdubbs] - Update to perl-5.36.1. Fixes #5246.

[bdubbs] - Update to linux-6.3.1. Fixes #5245.

[bdubbs] - Update to coreutils-9.3. Fixes #5244.

2023-04-15

[bdubbs] - Update to vim-9.0.1452. Addresses #4500.

[bdubbs] - Update to iana-etc-20230405. Addresses #5006.

[bdubbs] - Update to zstd-1.5.5. Fixes #5239.

[bdubbs] - Update to Python-3.11.3. Fixes #5240.

[bdubbs] - Update to meson-1.1.0. Fixes #5242.

[bdubbs] - Update to man-pages-6.04. Fixes #5238.

[bdubbs] - Update to linux-6.2.11. Fixes #5241.

2023-03-31

[xry111] - Update to linux-6.2.9 (security fix). Fixes #5230.

[xry111] - Update to grep-3.10. Fixes #5234.

[xry111] - Update to wheel-0.40.0. Fixes #5229.

[xry111] - Update to bc-6.5.0. Fixes #5228.

[xry111] - Update to texinfo-7.0.3. Fixes #5235.

[xry111] - Update to coreutils-9.2. Fixes #5232.

[xry111] - Update to libcap-2.68. Fixes #5236.

[xry111] - Update to tzdata-2023c. Fixes #5237.

[xry111] - Update to xz-5.4.2. Fixes #5233.

[xry111] - Update to openssl-3.1.0. Fixes #5227.

[xry111] - Add flit-core-3.8.0.

2023-03-15

[bdubbs] - Update to bc-6.4.0. Fixes #5217.

[bdubbs] - Update to grep-3.9. Fixes #5225.

[bdubbs] - Update to linux-6.2.6. Fixes #5226.

[bdubbs] - Update to iana-etc-20230306. Addresses #5006.

2023-03-04

[xry111] - Update to systemd-253. Fixes #5206.

[xry111] - Update to bc-6.3.1. Fixes #5217.

[xry111] - Update to linux-6.2.2 (security fixes). Fixes #5218.

[xry111] - Update to procps-ng-4.0.3. Fixes #5220.

[xry111] - Update to iproute2-6.2.0. Fixes #5221.

[xry111] - Update to meson-1.0.1. Fixes #5222.


install-kognoglfs.sh
#!/bin/bash

# KOGNOG LINUX FROM SCRATCH or KOGNOG LFS
# VERSION 3.0
# MANTAINER: BALIH KOGNOG
# CONTACT: balihkognog@gmail.com

### BASED ON ###

# LINUX FROM SCRATCH
# VERSION 12.0-systemd
# PUBLISHED SEPTEMBER 1st, 2023
# CREATED BY GERARD BEEKMANS
# MANAGING EDITOR: BRUCE DUBBS
# EDITOR: DOUGLAS R. RENO
# EDITOR: DJ LUCAS
# COPYRIGHT (c) 1999-2023, Gerard Beekmans
# All rights reserved.
# This book is licensed under a Creative Commons License.
# Computer instructions may be extracted from the book under the MIT License.
# Linux® is a registered trademark of Linus Torvalds.

# File: install-kognoglfs.sh
# Created on: 11/29/2023
# By: Balih Kognog
# Last Modified: 11/30/2023

#################
# Program Start #
#################

# while-menu-dialog: a menu driven system information program

DIALOG_CANCEL=1
DIALOG_ESC=255
HEIGHT=0
WIDTH=0

display_dialog() {
  dialog --title "$1" \
    --no-collapse \
    --msgbox "$result" 30 100
}

display_mainmenu() {
    while true; do
    exec 3>&1
    selection=$(dialog \
        --backtitle "KOGNOG LFS Ver. 3.0 | LINUX FROM SCRATCH BOOK AND INSTALLATION" \
        --title "Table of Contents" \
        --clear \
        --cancel-label "Exit" \
        --menu "Please select an option:" $HEIGHT $WIDTH 15 \
        "0" "STOP! - READ ME FIRST" \
        "." "Preface (optional - read it as many times as you want)" \
        "I" "Introduction" \
        "II" "Preparing for the Build" \
        "III" "Building the LFS Cross Toolchain and Temporary Tools" \
        "IV" "Building the LFS System" \
        "V" "Appendices" \
        2>&1 1>&3)
    exit_status=$?
    exec 3>&-
    case $exit_status in
        $DIALOG_CANCEL)
        clear
        echo "Program terminated."
        exit
        ;;
        $DIALOG_ESC)
        clear
        echo "Program aborted." >&2
        exit 1
        ;;
    esac
    case $selection in
        "0" )
        result=$(cat Documents/READ.ME)
        display_dialog "STOP! READ ME FIRST"
        ;;
        "." )
        # Calling the Preface Menu
        source Preface/menu-preface.sh
        ;;
        "I" )
        # Calling the Introduction Menu
        source ChapterI/menu-introduction.sh
        ;;
        "II" )
        result=$(echo "Opening Preparing for the Build ...")
        display_dialog "Preparing for the Build"
        ;;
        "III" )
        result=$(echo "Building the LFS Cross Toolchain and Temporary Tools ...")
        display_dialog "Building the LFS Cross Toolchain and Temporary Tools"
        ;;
        "IV" )
        result=$(echo "Opening Building the LFS System ...")
        display_dialog "Building the LFS System"
        ;;
        "V" )
        result=$(echo "Opening Appendices ...")
        display_dialog "Appendices"
        ;;
        
    esac
    done
}

# Showing Main Menu, until exit
display_mainmenu

Preface/menu-preface.sh
#!/bin/bash

# KOGNOG LINUX FROM SCRATCH or KOGNOG LFS
# VERSION 3.0
# MANTAINER: BALIH KOGNOG
# CONTACT: balihkognog@gmail.com

### BASED ON ###

# LINUX FROM SCRATCH
# VERSION 12.0-systemd
# PUBLISHED SEPTEMBER 1st, 2023
# CREATED BY GERARD BEEKMANS
# MANAGING EDITOR: BRUCE DUBBS
# EDITOR: DOUGLAS R. RENO
# EDITOR: DJ LUCAS
# COPYRIGHT (c) 1999-2023, Gerard Beekmans
# All rights reserved.
# This book is licensed under a Creative Commons License.
# Computer instructions may be extracted from the book under the MIT License.
# Linux® is a registered trademark of Linus Torvalds.

# File: menu-preface.sh
# Created on: 11/29/2023
# By: Balih Kognog
# Last Modified: 12/04/2023

#################
# Program Start #
#################

# while-menu-dialog: a menu driven system information program

DIALOG_CANCEL=1
DIALOG_ESC=255
HEIGHT=0
WIDTH=0

display_dialog() {
  dialog --title "$1" \
    --no-collapse \
    --msgbox "$result" 30 100
}

while true; do
  exec 3>&1
  selection=$(dialog \
    --backtitle "KOGNOG LFS Ver. 3.0 | PREFACE" \
    --title "Table of Contents" \
    --clear \
    --cancel-label "Previous Menu" \
    --menu "Please select an option:" $HEIGHT $WIDTH 15 \
    ".1" "About LFS Book Version 12 used fog Kognog LFS" \
    ".2" "About the Manteiner of Kognog LFS" \
    ".3" "Foreword" \
    ".4" "Audience" \
    ".5" "LFS Target Architectures" \
    ".6" "Prerequisites" \
    ".7" "LFS and Standards" \
    ".8" "Rationale for Packages in the Book" \
    ".9" "Typography" \
    ".10" "Structure" \
    ".11" "Errata and Security Advisories" \
    2>&1 1>&3)
  exit_status=$?
  exec 3>&-
  case $exit_status in
    $DIALOG_CANCEL)
      # Returning to the main menu
      display_mainmenu
      ;;
    $DIALOG_ESC)
      clear
      echo "Program aborted." >&2
      exit 1
      ;;
  esac
  case $selection in
    ".1" )
      result=$(cat Preface/About-lfs-book.txt)
      display_dialog "About LFS Book Version 12 used fog Kognog LFS"
      ;;
    ".2" )
      result=$(cat Preface/About-maintainer.txt)
      display_dialog "About the Manteiner of Kognog LFS"
      ;;
    ".3" )
      result=$(cat Preface/Foreword.txt)
      display_dialog "Foreword"
      ;;
    ".4" )
      result=$(cat Preface/Audience.txt)
      display_dialog "Audience"
      ;;
    ".5" )
      result=$(cat Preface/LFS-target-architectures.txt)
      display_dialog "LFS Target Architectures"
      ;;
    ".6" )
      result=$(cat Preface/Prerequisites.txt)
      display_dialog "Prerequisites"
      ;;
    ".7" )
      result=$(cat Preface/LFS-standards.txt)
      display_dialog "LFS and Standards"
      ;;
    ".8" )
      result=$(cat Preface/Packages-rationale.txt)
      display_dialog "Rationale for Packages in the Book"
      ;;
    ".9" )
      result=$(cat Preface/Typography.txt)
      display_dialog "Typography"
      ;;
    ".10" )
      result=$(cat Preface/Structure.txt)
      display_dialog "Structure"
      ;;
    ".11" )
      result=$(cat Preface/Errata-security.txt)
      display_dialog "Errata and Security Advisories"
      ;;
  esac
done

ChapterI/menu-introduction.sh
#!/bin/bash

# KOGNOG LINUX FROM SCRATCH or KOGNOG LFS
# VERSION 3.0
# MANTAINER: BALIH KOGNOG
# CONTACT: balihkognog@gmail.com

### BASED ON ###

# LINUX FROM SCRATCH
# VERSION 12.0-systemd
# PUBLISHED SEPTEMBER 1st, 2023
# CREATED BY GERARD BEEKMANS
# MANAGING EDITOR: BRUCE DUBBS
# EDITOR: DOUGLAS R. RENO
# EDITOR: DJ LUCAS
# COPYRIGHT (c) 1999-2023, Gerard Beekmans
# All rights reserved.
# This book is licensed under a Creative Commons License.
# Computer instructions may be extracted from the book under the MIT License.
# Linux® is a registered trademark of Linus Torvalds.

# File: menu-introduction.sh
# Created on: 12/04/2023
# By: Balih Kognog
# Last Modified: 12/04/2023

#################
# Program Start #
#################

# while-menu-dialog: a menu driven system information program

DIALOG_CANCEL=1
DIALOG_ESC=255
HEIGHT=0
WIDTH=0

display_dialog() {
  dialog --title "$1" \
    --no-collapse \
    --msgbox "$result" 30 100
}

while true; do
  exec 3>&1
  selection=$(dialog \
    --backtitle "KOGNOG LFS Ver. 3.0 | PREFACE" \
    --title "Table of Contents" \
    --clear \
    --cancel-label "Previous Menu" \
    --menu "Please select an option:" $HEIGHT $WIDTH 15 \
    ".1" "How to Build an LFS System" \
    ".2" "What's new since the las release of LFS" \
    ".3" "Changelog" \
    ".4" "Resources" \
    ".5" "Help" \
    2>&1 1>&3)
  exit_status=$?
  exec 3>&-
  case $exit_status in
    $DIALOG_CANCEL)
      # Returning to the main menu
      display_mainmenu
      ;;
    $DIALOG_ESC)
      clear
      echo "Program aborted." >&2
      exit 1
      ;;
  esac
  case $selection in
    ".1" )
      result=$(cat ChapterI/Howto-build.txt)
      display_dialog "How to Build an LFS System"
      ;;
    ".2" )
      result=$(cat ChapterI/Whats-new.txt)
      display_dialog "What's new since the las release of LFS"
      ;;
    ".3" )
      result=$(cat ChapterI/Changelog.txt)
      display_dialog "Changelog"
      ;;
    ".4" )
      result=$(cat ChapterI/Resources.txt)
      display_dialog "Resources"
      ;;
    ".5" )
      result=$(cat ChapterI/Help.txt)
      display_dialog "Help"
      ;;
  esac
done